From 4df9ad898ce02a8b343ecc500ebb1a579458fc0d Mon Sep 17 00:00:00 2001
From: Tiejun Chen <tiejun.chen@windriver.com>
Date: Thu, 31 Oct 2013 17:46:10 +0800
Subject: [PATCH 096/130] powerpc: Update KBUILD_AFLAGS with core specific
 options

Update also KBUILD_AFLAGS specifying the target core for CONFIG_E500
-mcpu=e5500/e500mc/8540
Add also -msoft-float.

Signed-off-by: Catalin Udma <catalin.udma@freescale.com>
Change-Id: Ibaf93105792b84dae836c7020ff708ed7afbad6a
Reviewed-on: http://git.am.freescale.net:8181/3764
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Wood Scott-B07421 <scottwood@freescale.com>
Reviewed-by: Rivera Jose-B46482 <Jose.G.Rivera@freescale.com>
[Fetch from http://git.freescale.com/git/cgit.cgi/ppc/sdk/linux.git/,
 Tag: fsl-sdk-v1.4.5
 rebase on current context.]
Signed-off-by: Tiejun Chen <tiejun.chen@windriver.com>
---
 arch/powerpc/Makefile |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/arch/powerpc/Makefile b/arch/powerpc/Makefile
index 9434c32..7b6613e 100644
--- a/arch/powerpc/Makefile
+++ b/arch/powerpc/Makefile
@@ -70,7 +70,7 @@ LDFLAGS_vmlinux	:= $(LDFLAGS_vmlinux-y)
 CFLAGS-$(CONFIG_PPC64)	:= -mminimal-toc -mtraceback=no -mcall-aixdesc
 CFLAGS-$(CONFIG_PPC32)	:= -ffixed-r2 -mmultiple
 KBUILD_CPPFLAGS	+= -Iarch/$(ARCH)
-KBUILD_AFLAGS	+= -Iarch/$(ARCH)
+KBUILD_AFLAGS	+= -msoft-float -Iarch/$(ARCH)
 KBUILD_CFLAGS	+= -msoft-float -pipe -Iarch/$(ARCH) $(CFLAGS-y)
 CPP		= $(CC) -E $(KBUILD_CFLAGS)
 
@@ -128,11 +128,14 @@ endif
 ifeq ($(CONFIG_E500),y)
 ifeq ($(CONFIG_64BIT),y)
 KBUILD_CFLAGS		+= -mcpu=e5500
+KBUILD_AFLAGS		+= -mcpu=e5500
 else
 ifeq ($(CONFIG_PPC_E500MC),y)
 KBUILD_CFLAGS		+= -mcpu=e500mc
+KBUILD_AFLAGS		+= -mcpu=e500mc
 else
 KBUILD_CFLAGS		+= -mcpu=8540
+KBUILD_AFLAGS		+= -mcpu=8540
 endif
 endif
 endif
-- 
1.7.5.4

